regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy

```diff
@@ -734,4 +734,17 @@ suite("test_date_function") {
            (
             SELECT FROM_UNIXTIME(UNIX_TIMESTAMP('20230918', '%Y%m%d'), 'yyyy-MM-dd HH:mm:ss') AS `a`
            )t """
+
+    sql """
+        create table date_varchar(
+            dt varchar null,
+            fmt varchar null
+        )
+        DISTRIBUTED BY HASH(`dt`) BUCKETS 1
+        properties("replication_num" = "1");
+    """
+    sql """ insert into date_varchar values ("2020-12-12", "%Y-%m-%d"), ("20201111", "%Y%m%d"), ("202012-13", "%Y%m-%d"); """
```

Comment:
also test some invalid values like 0000-00-00, and some very large values 9999-12-31

Reply:
we should define the valid range of input parameters and add some invalid parameter tests.

Reply:
added some. and more valid/invalid ranges has ensured in other origin tests